### PR TITLE
MSEARCH-299 Logging improvement

### PR DESCRIPTION
**Pull request by shorasul96**
## Purpose
_Implementing and improving logs for more productive troubleshooting_

## Approach
_Logging is implemented in business logic steps, most edge scenarios, conditional business logics and exceptional cases._

### Changes checklist
- [ ] API paths, methods, request or response bodies changed, added, or removed
- [ ] Database schema changes
- [ ] Interface versions changes
- [ ] Interface dependencies added, or removed
- [ ] Permissions changed, added, or removed
- [x] Check logging.

### CODE REVIEW

**Review by hasanepam — APPROVED** [review]

File: src/main/java/org/folio/search/service/converter/preprocessor/AuthorityEventPreProcessor.java [hunk 1]
```diff
@@ -61,15 +64,25 @@ public void init() {
...
-           ? getResourceEventsToUpdate(event)
-           : getResourceEvents(event, event.getType());
+    log.debug("process:: by [event: {}]", event);
+
+    if (event.getType() == ResourceEventType.UPDATE) {
+      log.debug("process:: eventType == UPDATE");
+      return getResourceEventsToUpdate(event);
+    }
+    log.debug("process:: eventType != UPDATE");
+    return getResourceEvents(event, event.getType());
```
Comment (viacheslavkol) [inline src/main/java/org/folio/search/service/converter/preprocessor/AuthorityEventPreProcessor.java:74]: ResourceEvent is too big to put it in logs. Only basic info could be logged (Same for other places where big object is loggged).
You shouldn't write multiple methods for different event types, you could just put eventType into a log

File: src/main/java/org/folio/search/service/metadata/LocalResourceProvider.java [hunk 2]
```diff
@@ -51,13 +52,13 @@ public Map<String, SearchFieldType> getSearchFieldTypes() {
...
   private void loadResourceDescriptions() {
     try {
+      log.debug("loadResourceDescriptions:: Attempting to load descriptions");
       this.resourceDescriptions = Stream.of(patternResolver.getResources(RESOURCE_DESCRIPTIONS_LOCATION_PATTERN))
         .filter(Resource::isReadable)
         .map(this::loadResourceDescription)
         .filter(Objects::nonNull)
         .toList();
     } catch (IOException e) {
-      log.error("Failed to read models [pattern: {}]", RESOURCE_DESCRIPTIONS_LOCATION_PATTERN);
```
Comment (viacheslavkol) [inline src/main/java/org/folio/search/service/metadata/LocalResourceProvider.java:60]: Why are this and next error logs removed?

File: src/main/java/org/folio/search/utils/LogUtils.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package org.folio.search.utils;
+
+import java.util.Collection;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class LogUtils {
```
Comment (viacheslavkol) [inline src/main/java/org/folio/search/utils/LogUtils.java:8]: I'd say it's not necessary. We usually log only size of list. And even list with size = 2 may be too big (if large objects stored) to put in log.
It should be decided individually when logging.

File: src/main/java/org/folio/search/service/IndexService.java [hunk 4]
```diff
@@ -47,15 +48,18 @@ public class IndexService {
...
       "Index cannot be created for the resource because resource description is not found.");
 
     var index = getIndexName(resourceName, tenantId);
     var settings = settingsHelper.getSettings(resourceName);
     var mappings = mappingHelper.getMappings(resourceName);
 
-    log.info("Creating index for resource [resource: {}, tenant: {}, mappings: {}, settings: {}]",
-      resourceName, tenantId, mappings, settings);
+    log.info("Attempts to create index by [indexName: {}, mappings: {}, settings: {}]",
+      index, mappings, settings);
```
Comment (viacheslavkol) [inline src/main/java/org/folio/search/service/IndexService.java:62]: We have multitenancy in system so It's needed to add tenantId, where it's available

**Review by viacheslavkol — COMMENTED** [review]
I believe we should add 'info' level logs for APIs' inputs/outputs to have some basic info in there.
Also I'd suggest returning removed warn level logs (or why were they removed?)

**Review by viacheslavkol — APPROVED** [review]

**Pull request merged**